File: hackerrank/easy/0011-plus-minus/js/solution.mjs
// solution.mjs

export function plusMinus(arr) {
  let pos = 0, neg = 0, zero = 0;
  for (const num of arr) {
    if (num > 0) pos++;
    else if (num < 0) neg++;
    else zero++;
  }
  const n = arr.length;
  console.log((pos / n).toFixed(6));
  console.log((neg / n).toFixed(6));
  console.log((zero / n).toFixed(6));
}

// --- demo run ---
plusMinus([-4, 3, -9, 0, 4, 1]);
// Expected:
// 0.500000
// 0.333333
// 0.166667

File: hackerrank/easy/0011-plus-minus/swift/solution.swift
// solution.swift

import Foundation

func plusMinus(arr: [Int]) {
    var pos = 0, neg = 0, zero = 0
    for num in arr {
        if num > 0 {
            pos += 1
        } else if num < 0 {
            neg += 1
        } else {
            zero += 1
        }
    }
    
    let n = Double(arr.count)
    let posRatio = Double(pos) / n
    let negRatio = Double(neg) / n
    let zeroRatio = Double(zero) / n
    
    print(String(format: "%.6f", posRatio))
    print(String(format: "%.6f", negRatio))
    print(String(format: "%.6f", zeroRatio))
}

// --- demo run ---
plusMinus(arr: [-4, 3, -9, 0, 4, 1])
// Expected output:
// 0.500000
// 0.333333
// 0.166667